pkg/registry/storage/csidriver/strategy.go

```diff
@@ -86,7 +86,9 @@ func (csiDriverStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.
 	// Any changes to the mutable fields increment the generation number.
 	if !apiequality.Semantic.DeepEqual(oldCSIDriver.Spec.TokenRequests, newCSIDriver.Spec.TokenRequests) ||
 		!apiequality.Semantic.DeepEqual(oldCSIDriver.Spec.RequiresRepublish, newCSIDriver.Spec.RequiresRepublish) ||
-		!apiequality.Semantic.DeepEqual(oldCSIDriver.Spec.SELinuxMount, newCSIDriver.Spec.SELinuxMount) {
+		!apiequality.Semantic.DeepEqual(oldCSIDriver.Spec.SELinuxMount, newCSIDriver.Spec.SELinuxMount) ||
+		!apiequality.Semantic.DeepEqual(oldCSIDriver.Spec.PodInfoOnMount, newCSIDriver.Spec.PodInfoOnMount) ||
+		!apiequality.Semantic.DeepEqual(oldCSIDriver.Spec.FSGroupPolicy, newCSIDriver.Spec.FSGroupPolicy) {
```

Comment:
Even if changing to the immutable fields? I did see the same logic in Job validation: https://github.com/kubernetes/kubernetes/blob/c1bbf791dd22e94aaa879f473ec9f10a24da0ec9/pkg/registry/batch/job/strategy.go#L138-L140

If it doesn't break anything, I agree to check any spec change here.

Reply:
I believe this is fine because even though `strategy.PrepareForUpdate` runs before `strategy.ValidateUpdate`, if you attempted to change an immutable field, validation would still fail the request, so the generation bump would not be persisted in that case.

The `Job` validation has the same pattern of immutable fields:

https://github.com/kubernetes/kubernetes/blob/27475f6cf45d0d0d7ed6fb41c360e6465513bc24/pkg/apis/batch/validation/validation.go#L388-L391